tests/components/xiaomi_miio/test_select.py

```diff
@@ -0,0 +1,229 @@
+"""The tests for the xiaomi_miio select component."""
+
+import enum
+import logging
+
+from arrow import utcnow
+import pytest
+
+from homeassistant.components.asuswrt.router import KEY_COORDINATOR
+from homeassistant.components.select.const import (
+    ATTR_OPTION,
+    ATTR_OPTIONS,
+    SERVICE_SELECT_OPTION,
+)
+from homeassistant.components.xiaomi_miio import (
+    UPDATE_INTERVAL,
+    _async_update_data_default,
+)
+from homeassistant.components.xiaomi_miio.const import (
+    CONF_DEVICE,
+    CONF_FLOW_TYPE,
+    CONF_MAC,
+    DOMAIN,
+    KEY_DEVICE,
+)
+from homeassistant.components.xiaomi_miio.select import (
+    MODEL_TO_ATTR_MAP,
+    SELECTOR_TYPES,
+    AttributeEnumMapping,
+    XiaomiMiioSelectDescription,
+)
+from homeassistant.const import ATTR_ENTITY_ID, CONF_MODEL
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from tests.common import MockConfigEntry, async_fire_time_changed
+from tests.components.xiaomi_miio import TEST_MAC
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class DummyEnum_0(enum.Enum):
+    """Enum to test auto naming."""
+
+    Option_0 = 0
+    Option_1 = 1
+
+
+class DummyEnum_1(enum.Enum):
+    """Enum to test predefined naming."""
+
+    BadName_0 = 0
+    BadName_1 = 1
+
+
+class DummySelectStatus:
+    """Dummy device status."""
+
+    def __init__(self):
+        """Initialize the dummy device status."""
+        self.dummy = DummyEnum_0.Option_1
+        self.bad_name = DummyEnum_1.BadName_1
+
+
+class DummySelectDevice:
+    """Dummy device."""
+
+    def __init__(self, model, *args, **kwargs):
+        """Initialize the dummy device."""
+        self.data = DummySelectStatus()
+
+    def status(self):
+        """Retrieve properties."""
+        return self.data
+
+    def set_dummy_value(self, value: DummyEnum_0):
+        """Set dummy enum value."""
+        self.data.dummy = value
+        return True
+
+    def set_bad_name_value(self, value: DummyEnum_1):
+        """Set bad name enum value."""
+        self.data.bad_name = value
+        return True
+
+
+@pytest.fixture(autouse=True)
+async def setup_select(hass: HomeAssistant):
+    """Initialize setup xiaomi_miio airhumidifier select entity."""
+
+    dummy_device = DummySelectDevice("Dummy")
+
+    MODEL_TO_ATTR_MAP["Dummy"] = [
+        AttributeEnumMapping("dummy", DummyEnum_0),
+        AttributeEnumMapping("bad_name", DummyEnum_1),
+    ]
+    SELECTOR_TYPES.append(
+        XiaomiMiioSelectDescription(
```

Comment:
Set up the integration instead and let it add an entity. Patch the library appropriately to avoid I/O. Then assert the state of the entity in the state machine `hass.states` and call services and assert mock calls to the patched library.

Reply:
Hi, @MartinHjelmare 
Thank you for feedback. I`m not very familiar with python. And now I know about patching in the tests, because of this PR. :). 
Could you looking in and told if now this is OK.